presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcConnectorFactory.java

```diff
@@ -40,9 +40,14 @@
 
     public JdbcConnectorFactory(String name, Module module)
     {
-        this(name, JdbcModuleProvider.withCredentialProvider(module));
```

Comment:
should we hide constructors (make them protected) of `JdbcConnectorFactory` and use static method instead? It's not obvious why `module` should be wrapped in `withCredentialProvider` and this commit changes mechanics of `JdbcConnectorFactory`

Reply:
Hiding constructors is quite invasive change.

You should use `withCredentialProvider` when connector expects user should pass `user` and `password` in config properties. 

It is a typical jdbc connector behaviour. Notice that typically connector extend `io.prestosql.plugin.jdbc.JdbcPlugin` and from that point of view nothing changes.